feat: 로그인 페이지 UI 구현 완료

woo-dev-log opened the pull request:
### 📌 PR 종류 (PR Type)
<!-- 아래 항목 중 해당하는 곳에 [x] 로 표시해주세요. -->

- [ ] 🐛 버그 수정 (Bugfix)
- [x] ✨ 기능 추가 (Feature)
- [x] 🎨 UI 스타일 변경 (Style) - CSS, 클래스명, 레이아웃 등 UI 변경
- [ ] ♻️ 코드 리팩토링 (Refactor) - 기능 변화 없이 코드 구조 개선
- [ ] 📦 패키지/빌드 관련 변경 (Build) - 설정 파일 수정, 의존성 추가 등
- [ ] 🧱 테스트 코드 추가/수정 (Test)
- [ ] 📝 문서 수정 (Docs)
- [ ] 🚧 WIP (작업 중 PR)
- [ ] 기타 (Other): <!-- 직접 입력 -->

---

### 📝 요약 (Summary)
<!-- 한두 문장으로 변경 내용을 요약해주세요. -->
로그인 페이지 UI 구현 - 모바일 최적화 레이아웃과 브랜드 디자인 적용

---

### 🔍 상세 내용 (Describe your changes)
<!-- 어떤 변경사항이 있었는지, 왜 변경했는지 상세히 작성해주세요. 코드 흐름, 주요 로직 등을 설명해도 좋습니다. -->
#### AuthLayout 컴포넌트 구현
- **로그인 전용 레이아웃 컴포넌트** 생성으로 GlobalLayout과 분리
- **480px 고정 레이아웃**으로 모바일 중심 디자인 적용
- **slate-100 외부 배경**으로 모바일 앱 시뮬레이션 효과 구현
- **브랜드 색상**(brand-background) 적용한 내부 컨테이너

#### 로그인 페이지 UI 구현
- **3단 구조 레이아웃**: 브랜드 소개 → 로고 → 로그인 버튼
- **타이포그래피 개선**: CO-MEET 로고 5xl, 부제목 lg로 시각적 계층 구성
- **여백 최적화**: mt-20으로 상단 여백 조정, 전체적인 비율 개선

#### GitHub 로그인 버튼 구현
- **GitHub 로고 통합**: 공식 GitHub 로고 SVG 파일 활용
- **justify-between 레이아웃**: 로고, 텍스트, 균형 요소 배치
- **브랜드 색상 적용**: brand-primary 배경에 brand-background 텍스트
- **호버 효과**: 90% 투명도로 부드러운 인터랙션 제공

#### 기술적 구현
- **shadcn/ui Button 컴포넌트** 기반 구현
- **브랜드 색상 시스템** 일관된 적용
- **의미적 HTML 구조** (section 태그 활용)

### 📸 스크린샷 (필수)
<!-- UI 변경이 있는 경우, 변경 전/후 비교나 주요 화면을 첨부해주세요. -->
| Before | After |
|--------|-------|
|        |<img width="1906" height="1069" alt="스크린샷 2025-07-19 162259" src="https://github.com/user-attachments/assets/6aaeddc3-9e9e-4e2c-a632-c9c08beee15b" />|
---

### ✅ 확인 사항 (CheckList)
<!-- PR을 보내기 전 스스로 점검할 수 있도록 합니다. 팀원이 리뷰할 때도 참고할 수 있습니다. -->
- [x] 브라우저에서 정상 동작 확인
- [x] 기존 기능에 영향 없음
- [x] 반응형 UI 대응 확인
- [x] 코드 컨벤션에 맞게 작성됨 (eslint, prettier 등)
- [x] 테스트 코드 추가/수정 또는 테스트 통과 확인

---
### 💬 기타 참고사항 (Optional)
<!-- 리뷰어가 알면 좋을 추가 정보, 요청 사항 등을 작성해주세요. -->
#### 주요 리뷰 포인트
#### 1. AuthLayout 컴포넌트화 검토 요청
**현재 구현**: 별도 `AuthLayout` 컴포넌트로 분리
```typescript
// 장점: GlobalLayout과 분리된 로그인 전용 레이아웃
// 단점: 현재 한 곳에서만 사용 (과도한 추상화 가능성)
```

**검토 사항**
- 현재는 **로그인 페이지 1곳**에서만 사용
- **단기 프로젝트**(4주)에서 컴포넌트 분리의 필요성
- **인라인 구현** vs **컴포넌트 분리** 중 어느 쪽이 더 적절한지 의견 요청

#### 2. GitHub 로고 사용에 대한 검토
**현재 구현**: 공식 GitHub 로고 SVG 파일 사용

**검토 사항**
- **로고 없이 텍스트만 사용**하는 기존 피그마 디자인과의 비교
- 사용자 경험 측면에서 **로고의 효과성**

#### 참고 자료
- https://github.com/logos

woo-dev-log committed:
feat: 로그인 페이지 UI 구현 완료

- AuthLayout 컴포넌트로 로그인 전용 레이아웃 생성
- GitHub 로고가 포함된 로그인 버튼 구현
- 모바일 최적화 480px 고정 레이아웃 적용
- 브랜드 색상 시스템 활용한 일관된 디자인

주요 구현 사항
- 3단 구조 레이아웃 (브랜드 소개, 로고, 로그인 버튼)
- GitHub 로고와 텍스트를 균형있게 배치한 버튼 디자인
- CO-MEET 브랜드 아이덴티티 반영한 색상 및 타이포그래피
- shadcn/ui Button 컴포넌트 기반 구현

Monixc commented:
### 버튼 아이콘 관련
버튼 내 아이콘 사용 좋습니다! 버튼 내 아이콘과 텍스트는 `justify-between`보다는 붙어있는 형태가 더 자연스럽다고 생각합니다. 나란히 배치하고, gap을 주는 방식은 어떤가요??

### 레이아웃 관련
`AuthLayout`을 별도로 분리하기보다는, GlobalLayout에서 showHeader, showFooter 플래그를 사용해서 조건부 렌더링하는 방식이 더 효율적이라고 생각하긴 합니다. 헤더, 푸터 표시 여부 말고는 차이가 없어서요...

하지만 지금 AuthLayout으로 분리해서 사용한 것도, 의도를 명확히 전달할 수 있어서 괜찮다고 봅니당. 수정은 필요없을 것 같아요!!

버튼 아이콘 위치만 수정해주실 수 있나요?

Monixc commented:
아, 사랑님께서 맡으신 온보딩 페이지도 헤더만 배치한 상황이라 `GlobalLayout`이랑 차이가 생기게 되네요. 이렇게 되면 온보딩용으로 또 별도 레이아웃 분리가 필요하게 될 수도 있어서 좀 비효율적으로 보입니다. `GlobalLayout` 에서 인라인으로 제어하는게 깔끔하겠네요. 번복해서 죄송하지만 인라인 방식으로 변경해주시면 좋을 것 같습니다!😓

woo-dev-log committed:
Merge branch 'develop' into feature/auth-login

woo-dev-log committed:
feat: 로그인 페이지 UI 완성 및 GlobalLayout 통합 구조 개선

- AuthLayout 제거하고 GlobalLayout으로 레이아웃 시스템 통합
- 옵션화된 GlobalLayout으로 Header/Footer 표시 제어 구현

woo-dev-log commented:
> 아, 사랑님께서 맡으신 온보딩 페이지도 헤더만 배치한 상황이라 `GlobalLayout`이랑 차이가 생기게 되네요. 이렇게 되면 온보딩용으로 또 별도 레이아웃 분리가 필요하게 될 수도 있어서 좀 비효율적으로 보입니다. `GlobalLayout` 에서 인라인으로 제어하는게 깔끔하겠네요. 번복해서 죄송하지만 인라인 방식으로 변경해주시면 좋을 것 같습니다!😓

### @Monixc 님의 피드백 반영완료 - 버튼 디자인 & GlobalLayout 옵션화 구조 개선
온보딩 페이지까지 고려하니 별도 레이아웃이 또 필요하게 되어서 비효율적이겠네요.

#### 1. 버튼 아이콘 배치 개선
- justify-between → gap을 활용한 자연스러운 나란히 배치  

#### 2. 레이아웃 구조 통합 및 옵션화
- AuthLayout 삭제하고 GlobalLayout을 옵션화하여 효율성 개선
- **추가 옵션**: `showHeader`, `showFooter`, `variant` 플래그

ksy90101 reviewed (commented):
👍
